### PR TITLE
Update EF Core config and delete repositories

### PR DESCRIPTION
**Pull request by bjss-chris**
Task: [AB#11053](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/11053)

### CODE REVIEW

File: src/NHSD.BuyingCatalogue.Ordering.Persistence/Data/ContactEntityTypeConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+{
+    internal sealed class ContactEntityTypeConfiguration : IEntityTypeConfiguration<Contact>
+    {
+        public void Configure(EntityTypeBuilder<Contact> builder)
+        {
+            builder.ToTable("Contact");
+            builder.Property(c => c.FirstName).HasMaxLength(100);
+            builder.Property(c => c.LastName).HasMaxLength(100);
+            builder.Property(c => c.Email).HasMaxLength(256);
+            builder.Property(c => c.Phone).HasMaxLength(35);
```
Comment (PhysicsJon) [inline src/NHSD.BuyingCatalogue.Ordering.Persistence/Data/ContactEntityTypeConfiguration.cs:15]: Can these not be done with annotations in the model?

File: src/NHSD.BuyingCatalogue.Ordering.Persistence/Data/OrderItemEntityTypeConfiguration.cs [hunk 2]
```diff
@@ -1,100 +1,69 @@
...
-            builder.Property(orderItem => orderItem.Created)
-                .HasColumnName(nameof(OrderItem.Created))
-                .HasCamelCaseBackingField(nameof(OrderItem.Created))
-                .IsRequired();
-
-            builder.Property(orderItem => orderItem.LastUpdated)
-                .HasColumnName(nameof(OrderItem.LastUpdated))
-                .HasCamelCaseBackingField(nameof(OrderItem.LastUpdated))
-                .IsRequired();
+                .HasColumnName("TimeUnitId");
```
Comment (PhysicsJon) [inline src/NHSD.BuyingCatalogue.Ordering.Persistence/Data/OrderItemEntityTypeConfiguration.cs:66]: Can these not be done with annotations in the model?

File: src/NHSD.BuyingCatalogue.Ordering.Persistence/Data/OrderItemRecipientEntityTypeConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            builder.HasOne(r => r.Recipient)
+                .WithMany()
+                .HasForeignKey(odsCode)
+                .OnDelete(DeleteBehavior.ClientSetNull)
+                .HasConstraintName("FK_OrderItemRecipients_OdsCode");
+
+            builder.HasOne<OrderItem>()
+                .WithMany(i => i.OrderItemRecipients)
+                .HasForeignKey(orderId, catalogueItemId)
+                .HasConstraintName("FK_OrderItemRecipients_OrderItem");
```
Comment (PhysicsJon) [inline src/NHSD.BuyingCatalogue.Ordering.Persistence/Data/OrderItemRecipientEntityTypeConfiguration.cs:35]: Can these not be done with annotations in the model?

File: src/NHSD.BuyingCatalogue.Ordering.Persistence/Data/OrderProgressEntityTypeConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    {
+        public void Configure(EntityTypeBuilder<OrderProgress> builder)
+        {
+            builder.ToTable("OrderProgress");
+            builder.Property(o => o.OrderId).ValueGeneratedNever();
+            builder.HasKey(o => o.OrderId);
+            builder.HasOne<Order>()
+                .WithOne(o => o.Progress)
+                .HasForeignKey<OrderProgress>(o => o.OrderId)
+                .HasConstraintName("FK_OrderProgress_Order");
```
Comment (PhysicsJon) [inline src/NHSD.BuyingCatalogue.Ordering.Persistence/Data/OrderProgressEntityTypeConfiguration.cs:17]: Can these not be done with annotations in the model?

File: src/NHSD.BuyingCatalogue.Ordering.Persistence/Data/OrderingPartyEntityTypeConfiguration.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+            builder.Property(o => o.OdsCode).HasMaxLength(8);
+            builder.HasOne(o => o.Address)
+                .WithMany()
+                .HasForeignKey("AddressId")
+                .OnDelete(DeleteBehavior.ClientSetNull)
+                .HasConstraintName("FK_OrderingParty_Address");
+
+            builder.HasIndex(o => o.OdsCode, "AK_OrderingParty_OdsCode")
+                .IsUnique()
+                .IsClustered();
```
Comment (PhysicsJon) [inline src/NHSD.BuyingCatalogue.Ordering.Persistence/Data/OrderingPartyEntityTypeConfiguration.cs:25]: Can these not be done with annotations in the model?

File: src/NHSD.BuyingCatalogue.Ordering.Persistence/Data/ServiceRecipientEntityTypeConfiguration.cs [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+namespace NHSD.BuyingCatalogue.Ordering.Persistence.Data
+{
+    internal sealed class ServiceRecipientEntityTypeConfiguration : IEntityTypeConfiguration<ServiceRecipient>
+    {
+        public void Configure(EntityTypeBuilder<ServiceRecipient> builder)
+        {
+            builder.ToTable("ServiceRecipient");
+            builder.HasKey(e => e.OdsCode);
+            builder.Property(e => e.OdsCode).HasMaxLength(8);
+            builder.Property(e => e.Name).HasMaxLength(256);
```
Comment (PhysicsJon) [inline src/NHSD.BuyingCatalogue.Ordering.Persistence/Data/ServiceRecipientEntityTypeConfiguration.cs:14]: Can these not be done with annotations in the model?

**Review by PhysicsJon — CHANGES_REQUESTED** [review]
A lot of the same comment

**Comment by bjss-chris** [comment]
> A lot of the same comment

We can use data annotations for some of this. However, there are certain things that we have to use the fluent API for. Therefore, back in the day we decided that we would only use the fluent API to keep things consistent.

I think historically there were a couple of data annotations lying around but the idea was to remove them (for the reason above, i.e. consistency).

**Comment by PhysicsJon** [comment]
> > A lot of the same comment
> 
> We can use data annotations for some of the context setup. However, there are certain things that we have to use the fluent API for. Therefore, back in the day we decided that we would only use the fluent API to keep things consistent.
> 
> I think historically there were a couple of data annotations lying around but the idea was to remove them (for the reason above, i.e. consistency).

That's fair, I'll amend my review

**Review by PhysicsJon — APPROVED** [review]

**Pull request merged**